### PR TITLE
Add measure for repository default branch as not `master`

### LINKED ISSUE #119: Implement measure for default branch naming as something besides `master`
Earlier versions of `git` used the name `master` for the default branch. This is considered offensive and could be exclusionary to some individuals ([see here](https://sfconservancy.org/news/2020/jun/23/gitbranchname/)). The Almanack could implement a check for this to make sure the default branch name is not `master` (perhaps with a suggestion towards `main`) following the pattern of GitHub and other git vendors.

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR adds a measure to check whether the default branch is not named `master` (can be named anything besides this).

Thanks for any feedback!

Closes #119 

## What is the nature of your change?

- [ ] Content additions or updates (adds or updates content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own contributions.
- [x] I have commented my content, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to related documentation (outside of book content).
- [x] My changes generate no new warnings.
- [x] New and existing tests pass locally with my changes.
- [x] I have added tests that prove my additions are effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: src/almanack/metrics/data.py [hunk 1]
```diff
@@ -166,6 +166,26 @@ def is_citable(repo: pygit2.Repository) -> bool:
...
+
+    Returns:
+        bool:
+            True if the default branch is "master", False otherwise.
+    """
+    # Get the symbolic reference to HEAD, which should point to the default branch
+    head_ref = repo.head.shorthand
+
+    # Check if the default branch is not "master"
+    return head_ref != "master"
```
Comment (gwaybio) [inline src/almanack/metrics/data.py:186]: don't most (and now default) repos use `main`? I'm wondering how this check might change in the future as more and more repos use main

File: src/almanack/metrics/metrics.yml [hunk 2]
```diff
@@ -59,6 +59,13 @@ metrics:
     description: >-
       Boolean value indicating the presence of a CITATION file
       or some other means of indicating how to cite the work.
+  - name: "default-branch-not-master"
```
Comment (gwaybio) [inline src/almanack/metrics/metrics.yml:62]: I'm wondering why this is a metric - are we making a value judgement on the name of the head reference?

**Review by gwaybio — APPROVED** [review]

File: src/almanack/metrics/data.py (hunk 1 above)
Comment (d33bs) [reply src/almanack/metrics/data.py:186]: `git` by default still references `master`. New settings were enabled to allow for a global override sometime in 2020 ([see here](https://github.blog/open-source/git/highlights-from-git-2-28/)) but `master` is still the embedded default without this being set. I double checked with this quick Dockerfile example https://gist.github.com/d33bs/dc9d2a8f64b792b73478e124db3e8601 and I see `master` when I run the container. There is a warning that the default is subject to change, yet it persists for now.

File: src/almanack/metrics/metrics.yml (hunk 2 above)
Comment (d33bs) [reply src/almanack/metrics/metrics.yml:62]: Thanks for this question! My guesstimate was that when cloning the repo that the default branch would be the head reference. I'm digging in to test this assumption out a bit and try to see if there's a way to find the default branch when working from a dev branch.

Comment (d33bs) [reply src/almanack/metrics/metrics.yml]: I ended up adding a check for the remote head and sought a remote `master` reference from there as this is a common pattern I found when looking through older GitHub repos.

**Comment by d33bs** [comment]
Thanks @gwaybio ! I've made a few changes based on your feedback - namely that we now look for a remote head and matching `master` reference. Merging this in now.

**Pull request merged**